src/lib/sfge/BaseSfgeEngine.ts

```diff
@@ -120,97 +126,77 @@ export class SfgeEngine extends AbstractRuleEngine {
 	/**
 	 * Helps make decision to run an engine or not based on the Rules, Target paths, and the Engine Options selected per
 	 * run. At this point, filtering should have already happened.
+	 * @override
 	 */
 	// eslint-disable-next-line @typescript-eslint/no-unused-vars
-	public shouldEngineRun(ruleGroups: RuleGroup[], rules: Rule[], target: RuleTarget[], engineOptions: Map<string,string>): boolean {
-		// If the engine isn't filtered out, there's no reason to not run it.
-		return true;
+	public shouldEngineRun(ruleGroups: RuleGroup[], rules: Rule[], target: RuleTarget[], engineOptions: Map<string, string>): boolean {
+		// If the EngineOptions doesn't include an SFGE config, don't try to run this engine.
+		// JUSTIFICATION: Validations earlier in the process prevent user from explicitly running
+		// SFGE with no config, but if SFGE is merely an enabled engine, then not specifying a config
+		// should implicitly cause SFGE to not run, instead of throwing a violation.
+		// Such an implementation is maximally courteous to users upgrading from a previous
+		// version.
+		if (engineOptions.has(CUSTOM_CONFIG.SfgeConfig)) {
+			return true;
+		} else {
+			uxEvents.emit(EVENTS.WARNING_VERBOSE, messages.getMessage("warning.sfgeSkippedWithoutConfig", []));
+			// TODO: We should throw a (verbose-only?) warning here, indicating that the engine
+			//  was implicitly skipped.
+			return false;
+		}
 	}
 
 	/**
 	 * @param engineOptions - A mapping of keys to values for engineOptions. Not all key/value pairs will apply to all engines.
+	 * @override
 	 */
-	public async run(ruleGroups: RuleGroup[], rules: Rule[], targets: RuleTarget[], engineOptions: Map<string,string>): Promise<RuleResult[]> {
+	public async run(ruleGroups: RuleGroup[], rules: Rule[], targets: RuleTarget[], engineOptions: Map<string, string>): Promise<RuleResult[]> {
 		// Make sure we have actual targets to run against.
 		let targetCount = 0;
 		targets.forEach((t) => {
 			if (t.methods.length > 0) {
-				// If we're targeting individual methods, then each method is counted as a separate target for this purpose.
+				// If we're targeting individual methods, then each method is
+				// counted as a separate target for this purpose.
 				targetCount += t.methods.length;
 			} else {
 				targetCount += t.paths.length;
 			}
 		});
 
+		// The rules have yet to be filtered by DFA/Non-DFA, so it's possible that some of the provided rules
+		// don't actually belong to this SFGE subvariant. So we should filter the rules so we only run ones
+		// in this engine's catalog.
+		const catalogRuleNames: Set<string> = new Set();
```

Comment:
Depending on the filter criteria, it's possible for the list of rules given to this method to contain both DFA and non-DFA rules. This lets us filter out the incompatible rules.